### PR TITLE
[MIG] subcontracted_service: Migration to 12.0

### PR DESCRIPTION
**Pull request by skukered**

### CODE REVIEW

**Review by HviorForgeFlow — APPROVED** [review]
Functional + Code review :+1:

File: subcontracted_service/models/warehouse.py [hunk 1]
```diff
@@ -43,11 +43,10 @@ def _get_vals_for_proc_rule_subcontracting(self):
     @api.multi
     def _set_subcontracting_service_proc_rule(self):
         for rec in self:
-            if rec.subcontracting_service_proc_rule_id:
-                continue
-            vals = rec._get_vals_for_proc_rule_subcontracting()
-            rule = self.env['stock.rule'].create(vals)
-            rec.subcontracting_service_proc_rule_id = rule.id
+            if not rec.subcontracting_service_proc_rule_id:
```
Comment (MiquelRForgeFlow) [inline subcontracted_service/models/warehouse.py:47]: I don't understand this change. Unnecessary diff. Please, revert it.

File: subcontracted_service/models/warehouse.py [hunk 2]
```diff
@@ -22,16 +22,7 @@ def _get_buy_route(self):
     def _get_vals_for_proc_rule_subcontracting(self):
         self.ensure_one()
         picking_type = self.in_type_id
-        if not picking_type:
```
Comment (MiquelRForgeFlow) [inline subcontracted_service/models/warehouse.py]: Please, could you explain why there is no need to validate picking_type? 🤔

Comment (HviorForgeFlow) [reply subcontracted_service/models/warehouse.py]: It was added due to this issue: https://github.com/OCA/purchase-workflow/issues/551

File: subcontracted_service/models/warehouse.py (hunk 1 above)
Comment (HviorForgeFlow) [reply subcontracted_service/models/warehouse.py:47]: It does the same without a continue.

**Review by HviorForgeFlow — COMMENTED** [review]
Are you planning to finish it?

File: subcontracted_service/models/warehouse.py (hunk 1 above)
Comment (MiquelRForgeFlow) [reply subcontracted_service/models/warehouse.py:47]: Yeah, I know. But it's unnecessary to add this code diff.

**Comment by skukered** [comment]
I'll check it out this week. Sorry

> Are you planning to finish it?

**Comment by skukered** [comment]
> Are you planning to finish it?

I'll check it out this week. Sorry

File: subcontracted_service/models/warehouse.py (hunk 1 above)
Comment (HviorForgeFlow) [reply subcontracted_service/models/warehouse.py:47]: 4 lines against code diff from README is minimalistic

**Comment by skukered** [comment]
"It does the same without a continue." @hveficent 
with the continue, I must add more code in the test to go through there

**Comment by LoisRForgeFlow** [comment]
@skukered  I think 97.95 codecov is enough. Could you squash all your migration commits into one?

**Review by JordiBForgeFlow — APPROVED** [review]
